https://issues.redhat.com/browse/ACM-10611 (#6292)
open-cluster-management/rhacm-docs: troubleshooting/trouble_postgresql.adoc

> [#troubleshooting-postgresql]

Reviewer: Similar to my previous comment about the context is not clear.

> [#troubleshooting-postgresql]

Author: ```suggestion
[#troubleshooting-postgresql-compliance-history]
```

> = Troubleshooting the PostgreSQL connection

Author: ```suggestion
= Troubleshooting the PostgreSQL connection to the compliance history API (Technology Preview) 
```

> When the RHACM hub has issues connecting to the PostgreSQL server, you may see Kubernetes events in the RHACM namespace with the following command:

Reviewer: ```suggestion
When the {product-title-short} hub has issues connecting to the PostgreSQL server, you may see events in the {product-title-short} namespace with the following command:
```

----
oc -n open-cluster-management get events --field-selector reason=OCMComplianceEventsDBError
----

The details are limited and will instruct you to view the Governance Policy Propagator logs for further details. To do so you may use the following command:

----
oc -n open-cluster-management logs -l name=governance-policy-propagator -f
----
 

In the logs, you may see an error message such as this, which indicates the password is not correctly specified.

2024-03-05T12:17:14.500-0500	info	compliance-events-api	complianceeventsapi/complianceeventsapi_controller.go:261	The database connection failed: pq: password authentication failed for user "rhacm-policy-compliance-history"

[#symptom-cluster-rotating-agents]

> == Symptom: Expired credentials for agents

Reviewer: This seems unrelated to ACM-10611.

Author: @mprahl it is, I was just creating a place holder and pulled a section from a previous topic. Still working on making updates. Thanks for your review and suggestions thus far

[#troubleshooting-postgresql-compliance-history]
= Troubleshooting the PostgreSQL connection to the compliance history API (Technology Preview) 

> The connection between PostgreSQL server and the policy compliance history API. For example, if you want to check for events where `reason=OCMComplianceEventsDBError` on your {product-title} hub cluster, run the following command:

Reviewer 2: The connection between PostgreSQL server and the policy compliance history API.

The suspense is killing me...

Reviewer 2: Same with tech preview tag, here.

[source,bash]
----
oc -n open-cluster-management get events --field-selector reason=OCMComplianceEventsDBError
----

> Limited details are provided and you are instructed to view the `governance-policy-propagator` logs for further details. Run the following command to check the `governance-policy-propagator` logs:

Reviewer 2: Limited details are provided. You can view the.....for more information.

[source,bash]
----
oc -n open-cluster-management logs -l name=governance-policy-propagator -f
----
 

> [#symptom-cluster-rotating-agents]

Author: ```suggestion
[#symptom-failed-postgresql-connection]
```

> == Symptom: Expired credentials for agents

Author: ```suggestion
== Symptom: PostgreSQL authentication fail
```

In the logs, you might see the following error message, which indicates the password is not correctly specified:

----
2024-03-05T12:17:14.500-0500	info	compliance-events-api	complianceeventsapi/complianceeventsapi_controller.go:261	The database connection failed: pq: password authentication failed for user "rhacm-policy-compliance-history"
----

> [#resolving-cluster-rotating-agents]

Author: ```suggestion
[#resolving-postgresql-server-connection]
```

> == Resolving the problem: Manually rotate credentials for agents

Author: ```suggestion
== Resolving the problem: PostgreSQL connection to the compliance history API (Technology Preview)
```

> //from the draft, its not as clear how the user can fix this. how can th user update the config for the connection?

Author: @mprahl will you help here
